src/endpoints/admin.rs

```diff
@@ -47,17 +55,26 @@ pub async fn post_admin(
     }
 
     if username != ARGS.auth_admin_username || password != ARGS.auth_admin_password {
-        return Ok(HttpResponse::Found()
-            .append_header(("Location", format!("{}/auth_admin/incorrect", ARGS.public_path_as_str())))
-            .finish());
+        return Ok((
+            StatusCode::FOUND,
+            [(
+                header::LOCATION,
+                format!("{}/auth_admin/incorrect", ARGS.public_path_as_str()),
+            )],
+            "".to_string(),
+        )
+            .into_response());
     }
 
-    let mut pastas = data.pastas.lock().unwrap();
+    let pastas = {
+        let mut pastas = data.pastas.lock().unwrap();
 
-    remove_expired(&mut pastas);
+        remove_expired(&mut pastas);
```

Comment:
another option for a nice PR, this should happen as soon as the pasta expires not only when a new one is pushed. Best way would be an async task or thread (I forgot if microbin uses threads or async tasks) which sleeps till the next pasta needs to be removed (or even simpler just try remove_expired every 10 seconds).

Reply:
Right. Should be quite easy doable. I'll create one when this pr is merged so we get not conflicts